src/main/webapp/app/code-editor/file-browser/code-editor-file-browser.component.ts

```diff
@@ -183,9 +201,17 @@ export class CodeEditorFileBrowserComponent implements OnChanges, AfterViewInit
      * @desc Calls the repository service to see if the repository has uncommitted changes
      */
     checkIfRepositoryIsClean = (): Observable<CommitState> => {
-        return this.repositoryService.isClean().pipe(
-            catchError(() => Observable.of(null)),
-            rxMap(res => (res ? (res.isClean ? CommitState.CLEAN : CommitState.UNCOMMITTED_CHANGES) : CommitState.COULD_NOT_BE_RETRIEVED)),
+        return this.repositoryService.getStatus().pipe(
+            rxMap(res => {
+                return res.repositoryStatus === 'CLEAN'
+                    ? CommitState.CLEAN
```

Comment:
This logic looks weird to me. Would a switch statement not make more sense to me? Why is the repositoryStatus not an enum as well?

Reply:
You are right, I also think this is strange in retrospect.
What happens here is that the Status from the server is mapped to client enum.
I changed and commented this part to make it more clear.